pkgs/by-name/me/mednaffe/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  autoreconfHook,
+  fetchFromGitHub,
+  gtk3,
+  mednafen,
+  pkg-config,
+  stdenv,
+  wrapGAppsHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
```

Comment:
Why `(finalAttrs: {` instead of just `rec {`?

Reply:
Because the majority of other packages use rec and achieve the same result as what wrapping the attributes in a function does.

Reply:
> Because the majority of other packages use rec

Popularity does not make a good argument here, or else things like treewide banishing unquoted URLs from the codebase would never happen.

> and achieve the same result as what wrapping the attributes in a function does

Although being spread over the whole Nixpkgs codebase, `rec` is an anti-pattern:

https://nix.dev/guides/best-practices#recursive-attribute-set-rec

Then, `finalAttrs` design pattern (a.k.a. __overlay style overridable recursive attributes__) was merged more than 3 years ago, to overcome some of the problems of `rec`.

> Also, what does this change fix or improve that required the removal of rec and wrapping it in a function?

#119942

Reply:
> Popularity does not make a good argument here, or else things like treewide banishing unquoted URLs from the codebase would never happen.

I am aware, however I believe the change should be made with a purpose.

> Although being spread over the whole Nixpkgs codebase, rec is an anti-pattern:
>
> https://nix.dev/guides/best-practices#recursive-attribute-set-rec
>
> Then, `finalAttrs` design pattern (a.k.a. overlay style overridable recursive attributes) was merged more than 3 years ago, to overcome some of the problems of `rec`.

Is there an RFC or issue about treewide removal of `rec`? The way it is upstream works fine currently. That nix.dev page recommends using `let...in` as an option.

Reply:
> I am aware, however I believe the change should be made with a purpose.

"Getting rid of `rec` (and other bad design patterns)" is a purpose, I suppose.

> Is there an RFC or issue about treewide removal of rec?

Reasoning backwards it is.

First, technically there is no need of a "treewide-rec-banish RFC" to banish `rec` from Nixpkgs.

Second, refactoring the expressions I maintain is in my discretion.

Third, objectively the code is not worse than before; it is still perfectly readable and understandable.

Fourth, and related to second, subjectively the code is better since it avoids pitfalls from bad design patterns, and it is better to me to maintain and improve it.

> The way it is upstream works fine currently.

Not a sufficient reason to hinder code refactoring.

> That nix.dev page recommends using let...in as an option.

Yes, and I do use it sometimes. 

But this nix.dev tutorial recommends let-in because, among other things, its purpose is to use more general approaches, especially in external codebases that will not necessarily have direct access to Nixpkgs.

Also, things like #119942 did not exist in that time.

Reply:
Alright, I understand. 👍